pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -286,34 +283,22 @@ func (r *MemberRollReconciler) reconcileObject(ctx context.Context, roll *maistr
 		memberStatusMap[c.Namespace] = c
 	}
 
-	// 3. create ServiceMeshMember object for each ns in spec.members
-	memberNamespaces := roll.Spec.Members
-	if roll.Spec.IsClusterScoped() {
-		nsList := &corev1.NamespaceList{}
-		err = r.Client.List(ctx, nsList)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-		memberNamespaces = []string{}
-		for _, ns := range nsList.Items {
-			if ns.Name != meshNamespace && !isExcludedNamespace(ns.Name) {
-				memberNamespaces = append(memberNamespaces, ns.Name)
+	// 2. create ServiceMeshMember object for each ns in spec.members
+	if mesh != nil {
+		for _, ns := range requiredNamespaces.List() {
+			member, err := r.ensureMemberExists(ctx, ns, mesh.Name, meshNamespace)
+			if err != nil {
+				return reconcile.Result{}, err
+			} else if member == nil {
```

Comment:
The previous branch of this if-else statement returns from function when it's true, so `else if` is not necessary in this case. What do you think about this?
```suggestion
			}
			if member == nil {
```

Reply:
You're correct, but the reason I did it like this is to clearly separate the exceptional and the normal outcome of the `ensureMemberExists` function. 

The exceptional outcomes of the function are an actual error or a nil member, both of which require special handling. By using `if ... else if` the handling of these outcomes is visually tied together, whereas a separate if statement wouldn't be (as much). 

In reality, the `ensureMemberExists` function should return a NamespaceNotExist type of error instead of returning a nil member.